### DISCUSSION TITLE
Extend default extractor to support Ruby shortcut for array of strings

### DISCUSSION THREAD

**Discussion by mdrbohlav** [tailwindlabs/tailwindcss#10653] [Help] [answered]
Hi everyone, I am unsuccessfully trying to figure out how to extend the default class names extractor to support the Ruby shortcut for array of strings.

Does anyone know how to do that or does anyone have a working solution for `.rb` and `.html.erb` files?

The shortcuts look like this:
```
%w[a b c]
 => ['a', 'b', 'c']
```
```
%w(a b c)
 => ['a', 'b', 'c']
```

**Comment by wongjn** [comment, marked as answer]
Just tried in [Tailwind Play](https://play.tailwindcss.com/EQll4Lrx2z) and Tailwind seems to pick the classes up.

**Reply by mdrbohlav** [reply]
Oh, I see. Great tool the playground, didn't know about that. I've tested it more and now I can see there is a bug with classes with a dot `.` in the name, eg. `.p-2.5`. Different situations:
- when the class is not the last one it is picked up but there is also `.p-2` class in the generated CSS ([playground](https://play.tailwindcss.com/fsa69Ly4Oc));
- when the class is the last one it is not picked up at all, only the wrong one `.p-2` ([playground](https://play.tailwindcss.com/8kwsN1uVWX));
- when there is a modifier used with the class the previous points apply as well but there is one more, when it is the only one class in the array the class `.\[md\:p-2\.5\]` is also generated ([playground](https://play.tailwindcss.com/SCpSPvilFv)).

**Reply by wongjn** [reply]
You could look at [transforming the file content string](https://tailwindcss.com/docs/content-configuration#transforming-source-files) that gets passed to Tailwind to perhaps remove the bracket syntax that Tailwind seems to be getting confused with.

**Reply by mdrbohlav** [reply]
Removing the special syntax instead of forcing tailwind to work with it is a good idea and works! Thank you for your help!
